language/tools/move-cli/src/package.rs

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, bail, Result};
+use include_dir::{include_dir, Dir};
+use once_cell::sync::Lazy;
+use std::{
+    collections::HashSet,
+    fs::{self, File},
+    io::Write,
+    path::Path,
+};
+
+use move_lang::{
+    compiled_unit::CompiledUnit, extension_equals, find_filenames, move_compile, path_to_string,
+};
+use std::path::PathBuf;
+use stdlib::{COMPILED_EXTENSION, MOVE_EXTENSION};
+use vm::file_format::CompiledModule;
+
+/// Directory name for the package source files under package/<name>
+const PKG_SOURCE_DIR: &str = "source_files";
+/// Directory name for the package binary files under package/<name>
+const PKG_BINARY_DIR: &str = "compiled";
+
+/// Content for the stdlib directory
+const DIR_STDLIB: Dir = include_dir!("../../stdlib/modules");
+/// Content for the nursery directory
+const DIR_NURSERY: Dir = include_dir!("../../stdlib/nursery");
+
+/// The portion in the stdlib directory that are considered stdlib
+const STDLIB_MODULE_FILES: [&str; 9] = [
+    "CoreAddresses.move",
+    "Errors.move",
+    "Event.move",
+    "FixedPoint32.move",
+    "Hash.move",
+    "LCS.move",
+    "Option.move",
+    "Vector.move",
+    "Signer.move",
+];
+
+/// Pre-defined stdlib package
+static PACKAGE_STDLIB: Lazy<MovePackage> = Lazy::new(|| MovePackage {
+    name: "stdlib",
+    sources: vec![
+        SourceFilter {
+            source_dir: &DIR_STDLIB,
+            inclusion: Some(STDLIB_MODULE_FILES.iter().copied().collect()),
+            exclusion: HashSet::new(),
+        },
+        SourceFilter {
+            source_dir: &DIR_NURSERY,
+            inclusion: None, // include everything
+            exclusion: HashSet::new(),
+        },
+    ],
+    deps: vec![],
+});
+
+static PACKAGE_LIBRA: Lazy<MovePackage> = Lazy::new(|| MovePackage {
+    name: "libra",
+    sources: vec![SourceFilter {
+        source_dir: &DIR_STDLIB,
+        inclusion: None, // include everything
+        exclusion: STDLIB_MODULE_FILES.iter().copied().collect(),
+    }],
+    deps: vec![&PACKAGE_STDLIB],
+});
+
+struct SourceFilter<'a> {
+    /// The embedded directory
+    source_dir: &'a Dir<'a>,
+    /// Source files to be included, if set to None, include everything
+    inclusion: Option<HashSet<&'a str>>,
+    /// Source files to be excluded, to exclude nothing, set it to empty
+    exclusion: HashSet<&'a str>,
```

Comment:
For consistency it might be best to make these both nops if they're the empty hashset, or they're both set to none. I have a personal preference for `None`, but I'm fine either way as long as they're consistent.

Reply:
I see the consistency argument. I can additionally make `exclusion` an Option as well and setting it to `None` will be the same as giving it an empty set. But that gives a bit inconsistency with the `None` for `inclusion` as well because a `None` in `inclusion` means include all but a `None` in `exclusion` means exclude nothing... So...

Reply:
Good point. I guess I was viewing it from the "what value means you should ignore this field's value" and the `None` made more sense that viewpoint, but I can see what you mean how it could also be confusing.

 I'm fine keeping it as-is if you would prefer to keep it this way, so feel free to land :)